resources/views/community/components/user/profile/primary-meta.blade.php

```diff
@@ -0,0 +1,81 @@
+@props([
+    'hardcoreRankMeta' => [],
+    'softcoreRankMeta' => [],
+    'userMassData' => [],
+    'username' => '',
+])
+
+<?php
+use App\Site\Enums\Permissions;
+use Illuminate\Support\Carbon;
+
+$hasVisibleRole = $userMassData['Permissions'] >= Permissions::JuniorDeveloper;
+$roleLabel = $hasVisibleRole ? Permissions::toString($userMassData['Permissions']) : '';
+$shouldMoveRoleToNextLine =
+    $hasVisibleRole
+    && ((mb_strlen($roleLabel) >= 12 && mb_strlen($username) >= 12) || mb_strlen($username) >= 16);
+?>
+
+<div class="flex flex-row-reverse sm:flex-row gap-x-4 mb-2">
+    <img
+        src="{{ media_asset('/UserPic/' . $username . '.png') }}"
+        alt="{{ $username }}'s avatar"
+        class="rounded-sm h-[64px] w-[64px] sm:max-h-[128px] sm:max-w-[128px] sm:min-w-[128px] sm:min-h-[128px] bg-stone-900 light:bg-neutral-100"
+    >
+
+    <div class="w-full">
+        <div class="flex sm:flex-row sm:justify-start sm:items-center gap-x-2 {{ $hasVisibleRole ? 'mb-2 sm:mb-0' : '' }} {{ $shouldMoveRoleToNextLine ? 'flex-col' : 'items-center' }}">
+            {{-- Username --}}
+            <h1 class='border-0 text-lg sm:text-2xl font-semibold mb-0'>{{ $username }}</h1>
+
+            {{-- Legacy Role --}}
+            {{-- TODO: Support N roles. --}}
+            @if ($hasVisibleRole)
+                <div class="flex h-4 items-center justify-center bg-neutral-700 text-neutral-300 px-1.5 rounded sm:-mt-1">
+                    <p class="text-2xs -mb-0.5">{{ $roleLabel }}</p>
+                </div>
+            @endif
+        </div>
+
+        {{-- Motto --}}
+        @if (!empty($userMassData['Motto']))
+            <div class="border-l border-text pl-2 italic text-2xs hyphens-auto mb-3">
+                <p style="word-break: break-word;">{{ $userMassData['Motto'] }}</p>
+            </div>
+        @endif
+
+        {{-- 🚨 Space is limited. Do NOT display more than 4 rows of content here. --}}
+        <div class="text-2xs">
+            {{-- Points --}}
+            <x-user.profile.points-display
+                :hardcorePoints="$userMassData['TotalPoints']"
+                :softcorePoints="$userMassData['TotalSoftcorePoints']"
+                :weightedPoints="$userMassData['TotalTruePoints']"
+            />
+
+            {{-- Site Rank --}}
+            <x-user.profile.site-rank-display
+                :hardcoreRankMeta="$hardcoreRankMeta"
+                :softcoreRankMeta="$softcoreRankMeta"
+                :userMassData="$userMassData"
+                :username="$username"
+            />
+
+            {{-- Last Activity --}}
+            <p>
+                <span class="font-bold">Last Activity:</span>
+                <span class="cursor-help" title="{{ getNiceDate(strtotime($userMassData['LastActivity'])) }}">
+                    {{ Carbon::parse($userMassData['LastActivity'])->diffForHumans() }}
+                </span>
+            </p>
```

Comment:
The [old implementation](https://github.com/RetroAchievements/RAWeb/pull/2175/files#diff-98777a5d0a4ac46a7efe8cada6437732a761b079f5b2f8d6221cf473b65cf7deL205) hid this field if a user didn't have any activity yet (see profile for newest created user)

Reply:
Thanks, resolved in latest.